### PR TITLE
Public Cloud: Fix Multi-machine terraform configuration for Azure

### PR DESCRIPTION
**Pull request by jlausuch**
Azure was never used before for multi-machine scenarios and it
was never tested properly. This fixes the problems we have
seen when applying terraform on Azure for more than 1 VM.

### CODE REVIEW

File: data/publiccloud/terraform/azure.tf [hunk 1]
```diff
@@ -92,16 +93,17 @@ resource "azurerm_network_security_group" "openqa-nsg" {
...
     network_security_group_id = "${azurerm_network_security_group.openqa-nsg.id}"
+    count                     = "${var.instance_count}"
 
     ip_configuration {
-        name                          = "openqa-nic-config"
+        name                          = "${element(random_id.service.*.hex, count.index)}-nic-config"
         subnet_id                     = "${azurerm_subnet.openqa-subnet.id}"
         private_ip_address_allocation = "dynamic"
-        public_ip_address_id          = "${azurerm_public_ip.openqa-publicip.id}"
+        public_ip_address_id          = ${element(azurerm_public_ip.openqa-publicip.*.id, count.index)}"
```
Comment (cfconrad) [inline data/publiccloud/terraform/azure.tf]:
```suggestion
        public_ip_address_id          = "${element(azurerm_public_ip.openqa-publicip.*.id, count.index)}"
```

File: data/publiccloud/terraform/azure.tf [hunk 2]
```diff
@@ -119,7 +121,7 @@ resource "azurerm_image" "image" {
 }
 
 resource "azurerm_virtual_machine" "openqa-vm" {
-    name                  = "${azurerm_resource_group.openqa-group.name}"
+    name                  = "${var.name}-${element(random_id.service.*.hex, count.index)}"
     location              = "${var.region}"
     resource_group_name   = "${azurerm_resource_group.openqa-group.name}"
     network_interface_ids = ["${azurerm_network_interface.openqa-nic.id}"]
```
Comment (cfconrad) [inline data/publiccloud/terraform/azure.tf]:
```suggestion
     network_interface_ids = ["${azurerm_network_interface.openqa-nic[count.index].id}"]
```

**Comment by asmorodskyi** [comment]
+1 to all points mentioned by @cfconrad , can not add anything more

**Comment by jlausuch** [comment]
@cfconrad Thanks for trying it out. I have amended the changes.

**Pull request merged**